docs/proposals/THV-2106-virtual-mcp-server.md

```diff
@@ -0,0 +1,1035 @@
+        audience: "github-api"
+        scopes: ["repo", "read:org"]
+        subject_token_type: "access_token"  # access_token | id_token
+
+    jira:
+      type: token_exchange
+      token_exchange:
+        token_url: "https://keycloak.example.com/realms/myrealm/protocol/openid-connect/token"
+        client_id: "vmcp-jira-exchange"
+        client_secret_env: "JIRA_EXCHANGE_SECRET"
+        audience: "jira-api"
+        scopes: ["read:jira-work", "write:jira-work"]
+
+    slack:
+      type: service_account
+      service_account:
+        credentials_env: "SLACK_BOT_TOKEN"
+        header_name: "Authorization"
+        header_format: "Bearer {token}"
+
+    internal-db:
+      type: pass_through  # Forward client token unchanged
+
+# ===== TOKEN CACHING =====
+token_cache:
+  provider: memory  # memory | redis | custom
+  config:
+    max_entries: 1000
+    ttl_offset: "5m"  # Refresh tokens 5 minutes before expiry
+
+# ===== TOOL AGGREGATION =====
+aggregation:
+  # Conflict resolution strategy
+  conflict_resolution: prefix  # prefix | priority | manual
+
+  # Conflict resolution details
+  conflict_resolution_config:
+    # For 'prefix' strategy: prefix format
+    prefix_format: "{workload}_"  # Options: {workload}, {workload}_, custom-prefix-
+
+    # For 'priority' strategy: explicit ordering
+    # priority_order: ["github", "jira", "slack"]
+
+    # For 'manual' strategy: must define overrides for all conflicts
+
+  # Tool filtering and overrides (per workload)
+  tools:
+    - workload: "github"
+      filter: ["create_pr", "merge_pr", "list_issues"]
+      overrides:
+        create_pr:
+          name: "gh_create_pr"
+          description: "Create a GitHub pull request"
+
+    - workload: "jira"
+      overrides:
+        create_issue:
+          name: "jira_create_issue"
+
+# ===== OPERATIONAL SETTINGS =====
+operational:
+  timeouts:
+    default: 30s
+    per_workload:
+      github: 45s
+
+  # Failure handling
+  failure_handling:
+    # Backend unavailability
+    health_check_interval: 30s
+    unhealthy_threshold: 3  # Mark unhealthy after N failures
+
+    # Partial failures
+    partial_failure_mode: fail  # fail | best_effort
+
+    # Circuit breaker
+    circuit_breaker:
+      enabled: true
+      failure_threshold: 5
+      timeout: 60s
+
+# ===== COMPOSITE TOOLS (Phase 2) =====
+composite_tools:
+  - name: "deploy_and_notify"
+    description: "Deploy PR with user confirmation and notification"
+    parameters:
+      pr_number: {type: "integer"}
+
+    # Workflow timeout
+    timeout: "30m"
+
+    steps:
+      - id: "merge"
+        tool: "github.merge_pr"
+        arguments: {pr: "{{.params.pr_number}}"}
+
+        # Per-step error handling
+        on_error:
+          action: "abort"  # abort | continue | retry
+
+      - id: "confirm_deploy"
+        type: "elicitation"
+        message: "PR {{.params.pr_number}} merged successfully. Proceed with deployment?"
```

Comment:
User-controlled template variables in elicitation messages could enable injection attacks? We'd need to be careful here